test/runtests.jl

```diff
@@ -2851,6 +2851,71 @@ end
 end
 end
 
+# ≈ doesn't recurse into tuples. this will also check both type and length implicitly
+tplapprox(a::Tuple, b::Tuple) = all(xy -> xy[1] ≈ xy[2], zip(a, b))
+
+@testset "Gradient and Jacobian Outputs" begin
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> 2*x, 3.0), (2.0,))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> 2*x, 3.0, Val(2)), (2.0,))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> [x, 2*x], 3.0), ([1.0,2.0],))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> [x, 2*x], 3.0, Val(2)), ([1.0,2.0],))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> sum(abs2, x), [2.0, 3.0]), (4.0, 6.0))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> sum(abs2, x), [2.0, 3.0], Val(2)), (4.0, 6.0))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> [-x[2], x[1]], [-2.0, 1.0]), ([0.0,1.0],[-1.0,0.0]))
+    @test tplapprox(Enzyme.unstructured_derivative(Enzyme.Forward, x -> [-x[2], x[1]], [-2.0, 1.0], Val(2)), ([0.0,1.0],[-1.0,0.0]))
+
+    x = 3.0
```

Comment:
for ease can we just call this scalar, and the other one vector, so they don't need overwriting?